crates/prover/src/adapter/memory.rs

```diff
@@ -33,28 +34,48 @@ impl From<crate::adapter::io::IoMemoryEntry> for MemoryEntry {
     }
 }
 
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+pub struct MemoryValue {
+    pub limb0: M31,
+    pub limb1: M31, // Always zero for felt values
+}
+
 #[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
 pub struct DataAccess {
     pub address: M31,
     pub prev_clock: M31,
-    pub prev_value: M31,
-    pub value: M31,
+    pub prev_value: MemoryValue,
+    pub value: MemoryValue,
 }
 
-#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub struct InstructionAccess {
+    pub instruction: Instruction,
     pub prev_clock: M31,
-    pub value: QM31,
 }
 
-#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub struct ExecutionBundle {
     pub registers: VmRegisters,
     pub clock: M31,
     pub instruction: InstructionAccess,
     pub operands: [Option<DataAccess>; 3],
 }
 
+impl Default for ExecutionBundle {
+    fn default() -> Self {
+        Self {
+            registers: VmRegisters::default(),
+            clock: M31::zero(),
+            instruction: InstructionAccess {
+                instruction: Instruction::Ret {},
```

Comment:
this is problematic as in store_fp_imm we do

```
        inputs.resize(1 << log_size, ExecutionBundle::default());
```

where we get operands[1] which is not defined.